Doc/library/stdtypes.rst

```diff
@@ -4822,35 +4822,66 @@ Generic Alias Type
    object: GenericAlias
    pair: Generic; Alias
 
-``GenericAlias`` objects are created by subscripting a class (usually a
-container), such as ``list[int]``.  They are intended primarily for
-:term:`type annotations <annotation>`.
-
-Usually, the :ref:`subscription <subscriptions>` of container objects calls the
-method :meth:`__getitem__` of the object.  However, the subscription of some
-containers' classes may call the classmethod :meth:`__class_getitem__` of the
-class instead. The classmethod :meth:`__class_getitem__` should return a
-``GenericAlias`` object.
+``GenericAlias`` objects are generally created by
+:ref:`subscripting<subscriptions>` a class. They are most often used with
+container classes (e.g. :class:`list` or :class:`dict`). They are intended
+primarily for use with :term:`type annotations <annotation>`. For example,
+``list[int]`` is a ``GenericAlias`` object created by subscripting the
+:class:`list` class with the argument :class:`int`.
+
+Usually, the :ref:`subscription <subscriptions>` of an object in Python will
+call the :meth:`__getitem__<object.__getitem__>` instance method defined on the
+object's class. For example, if we have a list
+``food = ['spam', 'eggs', 'bacon']``, calling ``food[0]`` will return the same
+value as calling ``type(food).__getitem__(food, 0)``. However, if a class
+defines the classmethod :meth:`__class_getitem__<object.__class_getitem__>`,
+then the subscription of that class may instead call the class's implementation
+of ``__class_getitem__``. This should return a ``GenericAlias`` object if it is
+properly defined. For example, because the :class:`list` class defines
+``__class_getitem__``, calling ``list[str]`` is equivalent to calling
+``list.__class_getitem__(str)``, rather than
+``type(list).__getitem__(list, str)``.
```

Comment:
Ahh, I see why you talked about ``type(food).__getitem__(food, 0)`` in such detail above. This is a nice parallel and definitely makes things easier to understand. I'd suggest extracting these bits resolving the confusion between ``__getitem__`` and ``__class_getitem__`` and place them in the docs for ``__class_getitem__`` instead https://docs.python.org/3/reference/datamodel.html#object.__class_getitem__ .

Reply:
Done!